src/graphql/mutations/UpdateUser.js

```diff
@@ -1,14 +1,64 @@
-import { GraphQLString, GraphQLNonNull } from 'graphql';
+import { GraphQLString } from 'graphql';
 import client from 'util/client';
 import User from 'graphql/models/User';
+import { omitBy } from 'lodash';
+import { AvatarTypes } from 'util/user';
+import AvatarTypeEnum from 'graphql/models/AvatarTypeEnum';
 
 export default {
   type: User,
   description: 'Change attribute of a user',
   args: {
-    name: { type: new GraphQLNonNull(GraphQLString) },
+    name: { type: GraphQLString },
+    slug: { type: GraphQLString },
+    avatarType: { type: AvatarTypeEnum },
+    avatarData: { type: GraphQLString },
+    bio: { type: GraphQLString },
   },
-  async resolve(rootValue, { name }, { userId }) {
+  async resolve(
+    rootValue,
+    { name, slug, avatarType, avatarData, bio },
+    { userId }
+  ) {
+    let doc = omitBy(
+      {
+        updatedAt: new Date().toISOString(),
+        name,
+        slug,
+        avatarType,
+        avatarData,
+        bio,
+      },
+      v => v === undefined || v === null || v === ''
+    );
+
+    if (Object.keys(doc).length === 1)
+      throw new Error(`There's nothing to update`);
+
+    // Ensure uniqueness of slug
+    if (slug !== undefined) {
+      const {
+        body: {
+          hits: { total },
+        },
+      } = await client.search({
```

Comment:
nit: if only count is needed, we can use [`client.count`](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_count) instead